src/python-bindings/htcondor2/credd.cpp

```diff
@@ -1 +1,175 @@
+bool
+_store_cred_failed(int result, int mode, const char ** errString) {
+    if( store_cred_failed(result, mode, errString) ) {
+        if( result == FAILURE ) { * errString = "Communication error"; }
+        return true;
+    }
+    return false;
+}
 
+bool
+cook_user(const char * user, int mode, std::string & cooked_user) {
+    std::string local;
+
+    if( user == NULL || user[0] == '\0' ) {
+        if(! (mode & STORE_CRED_LEGACY) ) {
+            return "";
+        }
+
+        char * uname = my_username();
+        local = uname;
+        free(uname);
+
+        char * dname = my_domainname();
+        if(dname != NULL) {
+            formatstr_cat( local, "@%s", dname );
+            free(dname);
+        } else {
+            // Is param() actually returning an allocated string here?
+            dname = param("UID_DOMAIN");
```

Comment:
Yes, it's returning a malloc()'d string here.

Reply:
Fixed.